### Expanded detection of *BAP1* alterations in cancer and tumor type-specific expression score comparison 

This is the Github project repository for the above-titled paper. \
Code for the project is available in the subdirectories here. \
Code is also available along with large file fize data on Zenodo at: https://zenodo.org/records/10180501

#### Abstract
*BAP1* is a tumor suppressor gene that was originally studied in uveal melanoma (UVM), kidney renal cell clear cell carcinoma (KIRC), and malignant mesothelioma (MESO). Early analyses focused on single-nucleotide variants, but other alteration types such as larger indels and gene-level copy number (CN) loss can also lead to loss of *BAP1* expression. We performed integrated multi-omic analyses using data from The Cancer Genome Atlas (TCGA) for 33 cancer types and more than 10,000 individuals. We combined and manually reviewed existing variant calls and new calls derived from a de novo local realignment pipeline across multiple independent variant callers including indel callers, increasing detection of high-quality somatic variant calls by 30% from 91 to 130, including 7 indels â‰¥40bp. Including CN loss alterations, 1561 samples from 32 cancer types were *BAP1*-altered, with alterations being predominantly CN-driven. Differential expression and survival analyses revealed both shared and tissue-specific consequences associated with BAP1 alteration. Our findings broadly emphasize the improvements that are gained by using new computational approaches in large cancer-genome studies such as TCGA.

